packages/@aws-cdk/aws-eks/README.md

```diff
@@ -811,6 +812,84 @@ const chart2 = cluster.addHelmChart(...);
 chart2.node.addDependency(chart1);
 ```
 
+#### CDK8s Charts
+
+[CDK8s](https://cdk8s.io/) is an open-source library that enables Kubernetes manifest authoring using familiar programming languages. It is founded on the same technologies as the AWS CDK such as [`constructs`](https://github.com/aws/constructs) and [`jsii`](https://github.com/aws/jsii).
+
+> To learn more about cdk8s, visit the [Getting Started](https://github.com/awslabs/cdk8s/tree/master/docs/getting-started) tutorials.
+
+The EKS module natively integrates with cdk8s and allows you to apply cdk8s charts on AWS EKS clusters via the `cluster.addCdk8sChart` method.
+
+To get started, add the `cdk8s` dependency to your `package.json` file.
+
+```json
+dependencies: {
+  "cdk8s": "^0.29.0"
+}
+```
+
+> Note that the version of `cdk8s` must be `>=0.29.0`.
+
+Next, install the `cdk8s-cli`:
+
+```console
+npm install -g cdk8s-cli
+```
+
+The CLI is only used to import the core kubernetes api objects, and is not necessarily required as a runtime dependency. In your project directory, run the following command:
+
+```console
+cdk8s import k8s@{kubernetes-cluster-version} -l typescript
+```
+
+For example, for `eks.KubernetesVersion.V1_17` you would run:
+
+```console
+cdk8s import k8s@1.17.0 -l typescript
+```
+
+This will create an `imports` directory that contains the entire k8s object API, as typescript constructs. We recommend you commit this directory as part of your source code.
+
+In your application code:
+
+```ts
+import * as eks from '@aws-cdk/aws-eks';
+import * as cdk8s from 'cdk8s';
+import * as k8s from './imports/k8s';
+
+const cluster = new eks.Cluster(this, 'HelloCDK8s', {
+  version: eks.KubernetesVersion.V1_17,
+});
+
+// create a cdk8s app
+const cdk8sApp = new cdk8s.App();
+
+// create a cdk8s chart
+const proxy = new cdk8s.Chart(cdk8sApp, 'ProxyChart');
+
+// create a pod using strongly typed k8s api.
+new k8s.Pod(proxy, 'Pod', {
```

Comment:
It would be worth documenting whether or not Stack resources that haven't been created yet can be referenced in a CDK8s chart: Ex: I want to create an S3 bucket and don't specify a name, and I want to inject into my Kubernetes chart that not-yet-created bucket name.

Reply:
Thanks for the callout, yes this will be possible.

I'll add it 👍